### PR TITLE
[WIP] [interp] Make the interpreter use the JIT EH code to perform exceptio…

### PR DESCRIPTION
**Pull request by vargaz**
…n handling.

This allows the removal of the duplicate and hard to understand interpreter EH code
and makes it easier to add debugger support to the intepreter.

### CODE REVIEW

File: mono/mini/mini-exceptions.c [hunk 1]
```diff
@@ -3195,6 +3213,7 @@ throw_exception (MonoObject *ex, gboolean rethrow)
 		for (l = ips; l; l = l->next) {
 			trace = g_list_append (trace, l->data);
 			trace = g_list_append (trace, NULL);
+			trace = g_list_append (trace, NULL);
```
Comment (kumpera) [inline mono/mini/mini-exceptions.c]: whar's this 3rd value?

Comment (vargaz) [reply mono/mini/mini-exceptions.c]: /* Only for interpreter frames */
	MonoJitInfo *ji;
}  ExceptionTraceIp;

Comment (vargaz) [reply mono/mini/mini-exceptions.c]: Its used by ves_icall_get_trace ().

**Comment by lewurm** [comment]
can we also get rid of the other `FILL_IN_TRACE` usages?

**Comment by alexrp** [comment]
Will this fix [#58437](https://bugzilla.xamarin.com/show_bug.cgi?id=58437)?

**Comment by vargaz** [comment]
Possibly.

**Comment by vargaz** [comment]
Closing in favor of #5902.

**Pull request closed without merging**